src/DeviceRenderer.js

```diff
@@ -83,23 +101,39 @@ module.exports = class DeviceRenderer {
 
     /**
      * Initialize custom plugins.
+     * @returns {Array} List of plugins to initialize.
      */
     addCustomPlugins() {
         const pluginInitMap = [
+            {enabled: this.options.touch, class: MultiTouchEvents},
+            {enabled: this.options.fullscreen, class: Fullscreen},
+            {enabled: this.options.clipboard, class: Clipboard, params: [this.options.i18n]},
+            {enabled: this.options.streamBitrate, class: StreamBitrate, params: [this.options.i18n]},
+            {enabled: this.options.camera, class: Camera, params: [this.options.i18n], dependencies:[MediaManager]},
+            {enabled: this.options.battery, class: Battery, params: [this.options.i18n]},
+            {enabled: this.options.gps, class: GPS, params: [this.options.i18n, this.options.gpsSpeedSupport]},
+            {enabled: this.options.capture, class: Screencast, params: [this.options.i18n]},
+            {enabled: this.options.streamResolution, class: StreamResolution},
+            {
+                enabled: this.options.buttons,
+                class: ButtonsEvents,
+                params: [this.options.i18n, this.options.translateHomeKey]
+            },
             {enabled: this.options.touch || this.options.mouse, class: CoordinateUtils},
             {enabled: this.options.keyboard, class: KeyboardEvents},
             {enabled: this.options.mouse, class: MouseEvents},
-            {enabled: this.options.gamepad, class: Gamepad, params: [this.gamepadManager, this.options.i18n]},
-            {enabled: this.options.camera, class: Camera, params: [this.options.i18n]},
+            {enabled: this.options.gamepad, class: Gamepad, params: [this.options.i18n], dependencies:[GamepadManager]},
```

Comment:
I see `GamepadManager` is not in this plugin map, nor the default one, except here as a dependency. Therefore it is loaded solely by the class name here. However the `MediaManager` is listed both directly as a dependency and as an item in the map.

I don't understand this inconsistency.

Reply:
Hum, not sure to correctly understand this part. I don't think this line `{enabled: this.options.microphone, class: MediaManager},` is require since microphone is only activate with camera, so the options.camera is enough.